src/ABLForcing/drivingForce/drivingForce.C

```diff
@@ -327,6 +333,36 @@ Type Foam::drivingForce<Type>::subtractVerticalPart_
 }
 
 
+template<class Type>
+void Foam::drivingForce<Type>::blendLinearSlopeDecay_
+(
+    List<Type>& source
+)
+{
+    Type slope0 = (source[hLevelBlend1] - source[hLevelBlend0])
+                / (  zPlanes_.planeLocationValues()[hLevelBlend1]
+                   - zPlanes_.planeLocationValues()[hLevelBlend0]);
+    Type dslope = -slope0
+                / (  zPlanes_.planeLocationValues()[hLevelBlendMax]
+                   - zPlanes_.planeLocationValues()[hLevelBlend1]);
+    // march from hLevelBlend1 (z >= maxAssimHeight)
+    // up to hLevelBlendMax (z >= maxAssimheight + blendThickness)
+    // as the slope decreases linearly to 0
+    for (label lvl=hLevelBlend1; lvl <= hLevelBlendMax; lvl++)
+    {
+        Type slope = slope0 + dslope*(  zPlanes_.planeLocationValues()[lvl]
```

Comment:
Note that at the last index of this for loop (lvl = hLevelBlendMax), the slope will be effectively zero. Therefore, the source term is constant starting at one grid level below the hLevelBlendMax level. In other words, you could also loop until lvl < hLevelBlendMax and start the second loop for setting the constant value at lvl = hlevelBlendMax. I don't think this will make a big difference, but I am not sure if you are aware.